src/__tests__/lib/containers/entity_finder/EntityFinder.test.tsx

```diff
@@ -0,0 +1,476 @@
+  Reference,
+} from '../../../../lib/utils/synapseTypes'
+
+jest.mock('../../../../lib/utils/hooks/SynapseAPI/useEntityBundle')
+jest.mock('react-reflex', () => {
+  return {
+    ReflexContainer: jest
+      .fn()
+      .mockImplementation(({ children }) => <div>{children}</div>),
+    ReflexElement: jest
+      .fn()
+      .mockImplementation(({ children }) => <div>{children}</div>),
+    ReflexSplitter: jest.fn().mockImplementation(() => <div></div>),
+  }
+})
+
+const TreeView = require('../../../../lib/containers/entity_finder/tree/TreeView')
+const DetailsList = require('../../../../lib/containers/entity_finder/details/EntityDetailsList')
+
+let invokeToggleSelectionViaTable: (reference: Reference) => void
+let invokeToggleSelectionViaTree: (reference: Reference) => void
+let invokeSetConfigViaTree: (
+  configuration: EntityDetailsListDataConfiguration,
+) => void
+
+TreeView.TreeView = jest
+  .fn()
+  .mockImplementation(({ toggleSelection, setDetailsViewConfiguration }) => {
+    invokeToggleSelectionViaTree = reference => {
+      toggleSelection(reference)
+    }
+    invokeSetConfigViaTree = configuration => {
+      setDetailsViewConfiguration(configuration)
+    }
+    return <div role="tree"></div>
+  })
+DetailsList.EntityDetailsList = jest
+  .fn()
+  .mockImplementation(({ toggleSelection }) => {
+    invokeToggleSelectionViaTable = (reference: Reference) => {
+      toggleSelection(reference)
+    }
+    return <div role="table"></div>
+  })
+
+const mockTreeView = TreeView.TreeView
+const mockDetailsList = DetailsList.EntityDetailsList
+
+jest.mock('../../../../lib/utils/SynapseClient', () => {
+  return {
+    getEntityPath: jest.fn(),
+    getEntityHeader: jest.fn(),
+    getEntityHeaders: jest.fn(),
+  }
+})
+const mockGetEntityHeaders = SynapseClient.getEntityHeaders as jest.Mock
+const mockUseGetEntityBundle = useGetEntityBundle as jest.Mock
+
+const mockOnSelectionChange = jest.fn()
+
+const defaultProps: EntityFinderProps = {
+  sessionToken: 'abcd',
+  initialScope: FinderScope.CURRENT_PROJECT,
+  projectId: 'syn456',
+  initialContainer: 'syn123',
+  selectMultiple: true,
+  onSelectedChange: mockOnSelectionChange,
+  showVersionSelection: true,
+  visibleTypesInList: [EntityType.FILE],
+  visibleTypesInTree: [EntityType.PROJECT, EntityType.FOLDER],
+  selectableTypes: Object.values(EntityType),
+  treeOnly: false,
+  selectedCopy: 'Chosen Entities',
+}
+
+function renderComponent(propOverrides?: Partial<EntityFinderProps>) {
+  return render(<EntityFinder {...defaultProps} {...propOverrides} />)
+}
+
+describe('EntityFinder tests', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+
+    mockUseGetEntityBundle.mockReturnValue({
+      data: {
+        entity: {
+          id: 'syn123',
+          name: 'My file entity',
+          concreteType: 'org.sagebionetworks.repo.model.FileEntity',
+        },
+      },
+    })
+  })
+
+  describe('single-select toggleSelection validation', () => {
```

Comment:
Defines the behavior for selecting, re-selecting, and selecting multiple versions of Entities